internal/service/loadbalancer/loadbalancer.go

```diff
@@ -89,17 +97,144 @@ func ResourceLoadBalancer() *schema.Resource {
 				Type:        schema.TypeString,
 				Computed:    true,
 			},
+			"nodes": {
+				Description: "Nodes are instances running load balancer service",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: loadBalancerNodeSchema(),
+				},
+			},
+			"network": {
+				Deprecated:  "Use 'networks' to define networks attached to load balancer",
+				Description: "Private network UUID where traffic will be routed. Must reside in load balancer zone.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+			},
 			"dns_name": {
+				Deprecated:  "Use 'networks' to get network DNS name",
 				Description: "DNS name of the load balancer",
 				Type:        schema.TypeString,
 				Computed:    true,
 			},
 		},
+		CustomizeDiff: customdiff.ForceNewIfChange("networks.#", func(ctx context.Context, old, new, meta interface{}) bool {
+			return new.(int) != old.(int)
+		}),
+	}
+}
+
+func loadBalancerNetworkSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Description: "The name of the network must be unique within the service.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.All(
+				validation.StringLenBetween(0, 65),
+				validation.StringMatch(regexp.MustCompile("^[a-zA-Z0-9_-]+$"), ""),
+			)),
+		},
+		"type": {
+			Description: "The type of the network. Only one public network can be attached and at least one private network must be attached.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			ValidateDiagFunc: validation.ToDiagFunc(
+				validation.StringInSlice([]string{
+					string(upcloud.LoadBalancerNetworkTypePrivate),
+					string(upcloud.LoadBalancerNetworkTypePublic),
+				}, false),
+			),
+		},
+		"family": {
+			Description: "Network family. Currently only `IPv4` is supported.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			ValidateDiagFunc: validation.ToDiagFunc(
+				validation.StringInSlice([]string{
+					string(upcloud.LoadBalancerAddressFamilyIPv4),
+				}, false),
+			),
+		},
+		"network": {
+			Description: "Private network UUID. Required for private networks and must reside in loadbalancer zone. For public network the field should be omitted.",
+			Type:        schema.TypeString,
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"dns_name": {
+			Description: "DNS name of the load balancer network",
+			Type:        schema.TypeString,
+			ForceNew:    true,
```

Comment:
Probably not needed, as this is a computed only attribute

Reply:
good catch, I'll fix that.